apps/aecore/lib/aecore/channel/channel_state_peer.ex

```diff
@@ -779,6 +818,10 @@ defmodule Aecore.Channel.ChannelStatePeer do
   Retrieves our offchain account balance from the latest offchain chainstate
   """
   @spec our_offchain_account_balance(ChannelStatePeer.t()) :: non_neg_integer()
+  def our_offchain_account_balance(%ChannelStatePeer{role: :delegate}) do
+    throw({:error, "#{__MODULE__}: Delagate doesn't have a balance"})
```

Comment:
Are the throws handled anywhere and can they be replaced with regular {:error, _} returns?

Reply:
@cytadela8

Reply:
This should only throw due to programmer error. It doesn't make sense to take a balance of delegate. It makes even less sens to get `foreign_offchain_account_balance` when ChannelStatePeer is for delegate.

Reply:
We use this functions only in places where it's know the ChannelStatePeer.role is not delegate.